test/e2e/e2e_test.go

```diff
@@ -0,0 +1,279 @@
+
+	// Ensure service plans were generated as well. Both classes and plans
+	// should have been generated by the broker.
+	plans, _ := svcatClient.ServicecatalogV1beta1().ServicePlans(namespace).List(metav1.ListOptions{})
+	assert.NotEmpty(t, plans.Items, "Expected service plans to exist")
+}
+
+// setupTest will create a new namespace for a single test and deploy the
+// broker inside.
+func setupTest(t *testing.T) string {
+	namespace := namespaceForTest(t)
+
+	_, err := kubeClient.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: namespace},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = deployBroker(namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = registerBroker(namespace)
+	if err != nil {
+		fmt.Println(err)
+		t.Fatal(err)
+	}
+
+	return namespace
+}
+
+// cleanupTest will destroy a test namespace and all its resources.
+func cleanupTest(t *testing.T) {
+	namespace := namespaceForTest(t)
+
+	err := kubeClient.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+	if err != nil {
+		panic(err)
+	}
+}
+
+// namespaceForTest will return a namespace name based on the current test.
+func namespaceForTest(t *testing.T) string {
+	return fmt.Sprintf("aosb-e2e-%s", strings.ToLower(t.Name()))
+}
+
+// deployBroker will deploy the broker inside the specified namespace.
+func deployBroker(namespace string) error {
+	numOfReplicas := int32(1)
+
+	// Create deployment of the broker server.
+	_, err := kubeClient.AppsV1().Deployments(namespace).Create(&appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &numOfReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": name,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						v1.Container{
+							Name:            name,
+							Image:           image,
+							ImagePullPolicy: "Never",
+							Ports: []v1.ContainerPort{
+								v1.ContainerPort{
+									ContainerPort: int32(4000),
+								},
+							},
+							Env: []v1.EnvVar{
+								v1.EnvVar{
+									Name:  "ATLAS_GROUP_ID",
+									Value: atlasGroupID,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_BASE_URL",
+									Value: atlasBaseURL,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_PUBLIC_KEY",
+									Value: atlasPublicKey,
+								},
+								v1.EnvVar{
+									Name:  "ATLAS_PRIVATE_KEY",
+									Value: atlasPrivateKey,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_USERNAME",
+									Value: basicAuthUsername,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_PASSWORD",
+									Value: basicAuthPassword,
+								},
+								v1.EnvVar{
+									Name:  "BROKER_HOST",
+									Value: "0.0.0.0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
```

Comment:
Can we use the definitions from the files in `/scripts/kubernetes/`? That way we can make sure our scripts always work.

Reply:
I think that should be possible but it would take some care as we have moved these to the samples folder and would most likely need some adjustments. Should I create a ticket for this or do you want it done in this PR?

Reply:
I'm okay with it being done as part of a different ticket in a separate PR. That being said, I'm not sure I understand what adjustments that would require. Is it not just a case of getting the contents of a file in a well known location and then deserialising it into Go structs?

Reply:
I was mostly thinking about the actual samples. As they will be used as part of the documentation as well it would take some thought to make sure they work properly in both scenarios.

Reply:
That's fair. Happy to do that as a separate ticket.

Reply:
Ticket created: https://jira.mongodb.org/browse/CLOUDP-48411